.github/workflows/ruby.yml

```diff
@@ -0,0 +1,43 @@
+name: Ruby
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:13
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: blogr_test
+        ports:
+        - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@473e4d8fe5dd94ee328fdfca9f8c9c7afc9dae5e
```

Comment:
Binding to an immutable SHA-1 may offer more reliability. But it does come with tradeoffs, as  hosted images toolsets (e.g. ubuntu-latest) move forward, and in the event of a breaking issue, actions may react with a patch to a major version to compensate. Meaning not binding to a major version (eg. ruby/setup-ruby@v1) may prevent you from taking advantage of bug fixes, critical functionality, and security fixes